CRM/Contact/Form/Task/PDFLetterCommon.php

```diff
@@ -100,6 +100,26 @@ public static function processMessageTemplate($formValues) {
     return [$formValues, $categories, $html_message, $messageToken, $returnProperties];
   }
 
+  /**
+   * Contact Form rule.
+   *
+   * @param array $fields
+   *   The input form values.
+   * @param array $files
+   * @param array $self
+   *   Additional values form 'this'.
+   *
+   * @return bool
+   *   TRUE if no errors, else array of errors.
+   */
+  public static function contactPdfFormRule($fields, $files, $self) {
```

Comment:
Do we need these much parameters i.e `$files, $self`. I would have thought we needed just the `field` parameter.

Also why is this function not in the `CRM_Contact_Form_Task_PDF` class? It's not like it is being used in this class.

Reply:
No we dont need all these but these are always provided while calling the form rule functions as you can see same has been defined for formRule function in CRM_Core_Form_Task_PDFLetterCommon class and most of the building of the form and its rules are defined in this class or its parent class